### ISSUE TITLE
--experimental_ui_max_stdouterr_bytes show head of the error

### ISSUE THREAD

**Issue by erenon** [bazelbuild/bazel#12602]
### Description of the problem / feature request:

By default `--experimental_ui_max_stdouterr_bytes=1048576` is in effect. If the output is bigger than that, the user gets back no output, forcing an another run, with the limit turned off, output piped to a file. This is inconvenient and slow: Instead, I'd like to see the beginning of the output (roughly up until the limit of stdouterr_bytes), to be able to make sense of the error.

### Feature requests: what underlying problem are you trying to solve with this feature?

To be able to easily deal with errors producing a lot of output (e.g: linker errors)

### What operating system are you running Bazel on?

Linux

### What's the output of `bazel info release`?

`3.6.0`

**Comment by michajlo** [comment]
Bazel should print the path of the file containing the dropped output, but that might not be in the version you're running.

We've resisted the partial output thing so far for performance and complexity reasons. Partial read cans be deceivingly expensive in certain implementations, and there are concerns about buffering too much output, especially in noisy builds. https://jmmv.dev/2020/09/bazel-ui-locking.html is a good writeup of how we landed where we are.
